packages/react-styled-ui/src/Scrollbar/Scrollbar.js

```diff
@@ -35,31 +41,35 @@ const Scrollbar = forwardRef((
   },
   ref,
 ) => {
+  const [isHydrated, setIsHydrated] = useState(false); // false for initial render
   const autoHeight = (maxHeight !== 'auto');
   const horizontalScrollbarVisibility = disabled ? 'hidden' : scrollbarVisibility;
   const verticalScrollbarVisibility = disabled ? 'hidden' : scrollbarVisibility;
-  let hideHorizontalTrackTimeout;
-  let hideVerticalTrackTimeout;
   let viewScrollLeft = 0;
   let viewScrollTop = 0;
   let lastViewScrollLeft = 0;
   let lastViewScrollTop = 0;
-  let isScrolling = false;
-  let isDragging = false;
-  let isTrackMouseOver = false;
-  let isViewMouseOver = false;
 
-  const [prevPageX, setPrevPageX] = useState(0);
-  const [prevPageY, setPrevPageY] = useState(0);
+  // For binding the `mousemove` and `mouseup` events to document, we use `useState` to store `startDragging` variable to trigger `useEffect`.
+  const [startDragging, setStartDragging] = useState(false);
+
+  // These variables are used to be the checked point to change DOM style directly (Do NOT need to re-render UI)
+  const isDraggingRef = useRef(false);
+  const isScrollingRef = useRef(false);
+  const isTrackMouseOverRef = useRef(false);
+  const isViewMouseOverRef = useRef(false);
+  const prevPageXRef = useRef(0);
+  const prevPageYRef = useRef(0);
 
   const viewRef = useRef(null);
   const trackHorizontalRef = useRef(null);
   const trackVerticalRef = useRef(null);
   const thumbHorizontalRef = useRef(null);
   const thumbVerticalRef = useRef(null);
 
+  const scrollbarWidth = getScrollbarWidth();
```

Comment:
Need to leave a comment to mention that `getScrollbarWidth` will access to DOM element, which is not safe for SSR. It's better to calculate the scrollbar width after the first render using the useEffect Hook.

Reply:
Thanks @cheton! Fixed in [58793a3]